### DISCUSSION TITLE
Discussion points ice micro with ICON devs

### DISCUSSION THREAD

**Discussion by bartvstratum** [microhh/microhh#84] [General]
Overview of things to discuss with Axel/Alberto about SB06 microphysics.

**Comment by bartvstratum** [comment]
Use of `q_crit` limiters. Some routines use if (`qr > q_crit`) or if (`qc > q_crit`), but not consistently. For example, `q_crit` is used in autoconversion (`qc > q_crit`) and sedi_vel_rain (`qr > q_crit`), but not in accretion (which uses `qr > 0` and `qc > 0`), selfcollection (which uses `qr > 0`), evaporation (`qr > 0`, `qc < q_crit`), etc. Is this done deliberately to prevent having left over rain which is not remove by any process?

**Comment by bartvstratum** [comment]
Use decreased sedimentation velocity in clouds? This seems to be the default in ICON, but causes discontinuities in qr/Nr at clouds base.

**Comment by bartvstratum** [comment]
What (exactly..) should we do with the saturation adjustment?

**Reply by bartvstratum** [reply]
For the setup with prognostic ice, we currently satadjust everything to `ql`, and let microphysics convert it to `qi`?

**Comment by bartvstratum** [comment]
What to do with `qsat()`? It is currently calculated as `f * qsat_liq() + (1-f) * qsat_ice()`, with `f` a function of temperature (probably following NICAM?). Should we calculate `f` as a function of the saturation adjusted `qc` and prognostic `qi`?

**Reply by bartvstratum** [reply]
At the moment (when using prognostic ice), `qsat()` is simply `qsat_liq()`.

**Comment by bartvstratum** [comment]
The number densities are limited where `qx > 0 && nx < eps` for rain, ice, snow, graupel, but not for hail? There is a `set_qnh` function, but it is not called from `set_default_n` in ICON.

**Reply by MirjamTijhuis** [reply]
In the 2024-01 open-source release of ICON,nh is limited with set_qnh_expPSD_N0const. I added this one to MicroHH.

**Comment by bartvstratum** [comment]
Most kernels calculate a tendency, and then multiply it by `dt` to get a total change (since ICON integrates the tendencies in each kernel), but not all. For example, none of the ice nucleation code has a `dt` in it. That would mean that as long as not all vapor is instantaneously converted to ice, this conversion in the model is happening faster if the time step is small, because per given unit of time, the code is called more often?

**Comment by MirjamTijhuis** [comment]
How is water conserved in ice_riming and snow_riming? It seems like the rime rates of different species can differ.

**Comment by MirjamTijhuis** [comment]
The homogenous freezing part of the cloud_freeze is impossible to reach. Is that on purpose?
